eng/NoTargetsSdk.AfterProps.props

```diff
@@ -0,0 +1,8 @@
+<Project>
+
+  <PropertyGroup>
+    <!-- NoTargets SDK needs a TFM set. So we used the latest .NETCoreApp supported one. -->
+    <TargetFramework Condition="'$(TargetFramework)' == ''">$(NetCoreAppToolCurrent)</TargetFramework>
```

Comment:
Should this only be set if `TargetFrameworks` is not set?  IOW does the NoTargets SDK support cross-targeting?

Reply:
Unfortunately at that point in the evaluation, neither TargetFramework nor TargetFrameworks is available unless they are set in a Directory.Build.props file (which is less likely).

Reply:
I see, in that case will this break cross-targeting since `TargetFramework` is the propety used to differentiate between  inner/outer builds?  Do we ever rely on cross-targeting in projects that would use this?

Reply:
The Traversal SDK doesn't allow specifying `TargetFrameworks` but the NoTargets SDK does allow outer and inner builds. That said, our three usages (observable in this PR) of `TargetFrameworks` in NoTargets projects are actually misuses:

1. externals.csproj specifies TargetFrameworks as it restores RID specific assets. That whole project should go away in the near future in favor of live built assets.
2. System.Composition.csproj which funnily enough is titled csproj even though it doesn't compile any C# project is using `TargetFrameworks` as well even though it's not clear to me why. My guess is so that traversal filtering works even though I don't think that's necessary.
3. Microsoft.Extensions.HostFactoryResolver.Sources is packaging a single tfm but it using TargetFrameworks. I highly doubt that's necessary as there isn't any runtime specific asset involved.

Theoretically someone could intentionally add a project that multi-targets and uses the NoTargets SDK but that's very unlikely. In such case the project would need to clear the TargetFramework value set by this file:

```
<TargetFramework />
```

Reply:
I see, so long as you're confident we don't need that for our usage I'm OK with this.

Reply:
@ericstj I found a way to check on TargetFrameworks as well as the NoTargets SDK offers a more suitable extension point: https://github.com/microsoft/MSBuildSdks/blob/7af5df96271f01d9a47383489283f7e90e435e9e/src/NoTargets/Sdk/Sdk.targets#L22